theano/tensor/tests/test_opt.py

```diff
@@ -3727,6 +3725,22 @@ def test_xor(self):
         self.assert_eqs_const(f, 0)
 
 
+    def test_stacktrace(self):
+        mode = theano.compile.get_default_mode().including(
+            'local_useless_elemwise_comparison')
+
+        x = T.vector('x', dtype=config.floatX)
+        f = theano.function([x], T.gt(x, x), mode=mode)
+        self.assertTrue(check_stack_trace(f, ops_to_check='last'))
+
+        f = theano.function([x], T.le(x, x), mode=mode)
+        self.assertTrue(check_stack_trace(f, ops_to_check='last'))
+
+        # Julian: I tried testing the stack trace for a bunch of different
+        # functions, including maximum and shapes, but other opts remove
+        # the stack traces in this case.
```

Comment:
Indeed, in the case you mention, only the `TensorConstant{0}` remains in the graph, and since it is a scalar constant, it it re-used everywhere and does not have a stack trace of its creation. Then the DeepCopyOp gets inserted to avoid returning internal memory to the user.
In any case, there is no node created by the optimization in this case, so there is no stack trace to look for. I was mistaken about that in my comment above.
It may be the same for the other csae, since again we are returning a scalar 0, not an alloc(0, ...).